url/resources/urltestdata.json

```diff
@@ -3778,6 +3778,81 @@
     "search": "",
     "hash": ""
   },
+  {
+    "input": "non-special:opaque  ",
+    "base": null,
+    "href": "non-special:opaque",
+    "origin": "null",
+    "protocol": "non-special:",
+    "username": "",
+    "password": "",
+    "host": "",
+    "hostname": "",
+    "port": "",
+    "pathname": "opaque",
+    "search": "",
+    "hash": ""
+  },
+  {
+    "input": "non-special:opaque  ?hi",
+    "base": null,
+    "href": "non-special:opaque?hi",
+    "origin": "null",
+    "protocol": "non-special:",
+    "username": "",
+    "password": "",
+    "host": "",
+    "hostname": "",
+    "port": "",
+    "pathname": "opaque",
+    "search": "?hi",
+    "hash": ""
+  },
+  {
+    "input": "non-special:opaque  #hi",
+    "base": null,
+    "href": "non-special:opaque#hi",
+    "origin": "null",
+    "protocol": "non-special:",
+    "username": "",
+    "password": "",
+    "host": "",
+    "hostname": "",
+    "port": "",
+    "pathname": "opaque",
+    "search": "",
+    "hash": "#hi"
+  },
```

Comment:
It might be useful to add a copy of these with a trailing `%20` (which should not be stripped)